index.js

```diff
@@ -10,7 +10,10 @@ module.exports = {
           * Always delay requires, otherwise your plugin will cause trouble
           * with db-migrates performance and generates issues to your users.
           */
-        require('typescript').register();
+        var typescriptModule = require('typescript');
+        if (typescriptModule && typescriptModule.register) {
```

Comment:
This is ok for me, but we should place a note to come back to this if there is ever support added for the register method.